### PR TITLE
Added ingesting users, vulnerabilities, scans, assets

### PR DESCRIPTION
**Pull request by befirst**

### CODE REVIEW

File: src/tenable/TenableClient.ts [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+  public async fetchReportByImageDigest(digestId: string): Promise<Report> {
+    const reportResponse = (await this.makeRequest(
+      `/container-security/api/v1/reports/by_image_digest?image_digest=${digestId}`,
+      Method.GET,
+      {},
+    )) as ReportResponse;
+    return reportResponse;
+  }
+
+  private async makeRequest(
```
Comment (Andrew8xx8) [inline src/tenable/TenableClient.ts]: use type arguments

File: src/tenable/fetchTenableData.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  ContainerVulnerability,
+  Dictionary,
+  Report,
+  Scan,
+  ScanDetail,
+  TenableDataModel,
+  WebAppVulnerability,
+} from "./TenableClient";
+
+export default async function fetchTenableData(
```
Comment (Andrew8xx8) [inline src/tenable/fetchTenableData.ts]: split this function

File: src/tenable/TenableClient.test.ts [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    expect(scan).not.toEqual({});
+  });
+
+  test("fetchVulnerabilities ok", async () => {
+    const client = await getClient();
+
+    const { nockDone } = await nock.back("vulnerabilities-ok.json", {
+      before: prepareScope,
+    });
+    const scans = await client.fetchScans();
```
Comment (Andrew8xx8) [inline src/tenable/TenableClient.test.ts]: Test only one method

File: src/utils/getUnique.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+export default function getUnique(array: any[], uniqueField: string) {
```
Comment (Andrew8xx8) [inline src/utils/getUnique.ts]: use `${host_id}_${plugin_id}` as a key

**Comment by aiwilliams** [comment]
We need to change the class value on a couple things, but I'll create an Issue once we've got this into our development environment and connected to a Tenable account, so that we can learn more about reports, scans, malware, etc.

**Pull request merged**